classes/publication/Repository.php

```diff
@@ -661,6 +686,66 @@ public function delete(Publication $publication)
         Hook::call('Publication::delete', [&$publication]);
     }
 
+    /**
+     * Given a Version Stage and a flag of whether the Version isMinor, 
+     * the publication's related data is being updated
+     *
+     * @hook 'Publication::updateVersionData::before' [[ &$newPublication, $publication ]]
+     */
+    public function updateVersionData(Publication $publication, VersionStage $versioningStage, bool $isMinor = true): Publication
+    {
+        $submission = Repo::submission()->get($publication->getData('submissionId'));
+        $nextAvailableVersionStage = Repo::submission()->getNextAvailableVersionData($submission, $versioningStage, $isMinor);
+
+        $newPublication = clone $publication;
+        $newPublication->setData('versionIsMinor', $isMinor);
+        $newPublication->setVersionData($nextAvailableVersionStage);
+
+        $newPublication->stampModified();
+        Hook::call(
+            'Publication::updateVersionData::before',
+            [
+                &$newPublication,
+                $publication
+            ]
+        );
+
+        $this->dao->update($newPublication);
+
+        $newPublication = Repo::publication()->get($newPublication->getId());
+
+        return $newPublication;
+    }
+
+    /**
+     * Get the string that describes the 
+     * given publication's version.
+     */
+    public function getVersionDataDisplay(Publication $publication, ?Submission $submission = null, ?Context $submissionContext = null): string 
```

Comment:
This feels like something that would be better coded in the front end VueJS -- it's a lot of boilerplate to make sure we have the necessary objects, then a call to get a translation. This might be necessary until we have a front end in place, but I suspect it'll be unused after that.

Reply:
@asmecher I agree that this could be better handled in the front end. I am leaving it for now but we can keep that in mind through the UI development part of this feature, and deprecate it when/if needed. 

Also I would like to remove the `versionString` property from the `publication.json` schema description - I will leave it for now in order to accommodate the access for the version display property from the UI and the templates. But I will keep in mind that it could be removed by the end of this feature.